cl/users/tests.py

```diff
@@ -905,3 +909,646 @@ def test_update_ban_object(self) -> None:
         # Checks email ban is updated with the hard bounce subtype
         self.assertEqual(email_ban.count(), 1)
         self.assertEqual(email_ban[0].event_sub_type, SUB_TYPES.GENERAL)
+
+
+class CustomBackendEmailTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = UserFactory()
+        test_dir = Path(settings.INSTALL_ROOT) / "cl" / "users" / "test_assets"
+        with open(test_dir / "hard_bounce.json", encoding="utf-8") as file:
+            cls.hard_bounce_asset = json.load(file)
+        with open(
+            test_dir / "general_soft_bounce.json", encoding="utf-8"
+        ) as file:
+            cls.soft_bounce_asset = json.load(file)
+        with open(
+            test_dir / "msg_large_bounce.json", encoding="utf-8"
+        ) as file:
+            cls.soft_bounce_msg_large_asset = json.load(file)
```

Comment:
Just FYI, you can combine this in Python 3.10 to one big with statement:

```suggestion
        with (
            open(test_dir / "hard_bounce.json", encoding="utf-8") as hard_bounce,
            open(test_dir / "general_soft_bounce.json", encoding="utf-8") as soft_bounce,
            open(test_dir / "msg_large_bounce.json", encoding="utf-8") as large_bounce,
        ):
            cls.hard_bounce_asset = json.load(hard_bounce)
            cls.soft_bounce_asset = json.load(soft_bounce)
            cls.soft_bounce_msg_large_asset = json.load(large_bounce)
```

https://docs.python.org/3/whatsnew/3.10.html#parenthesized-context-managers

A little nicer. :)

Reply:
I've changed this and also in the previous phase tests assets to use parenthesized-context-managers, however, seems there is a problem with Black to parse this new syntaxis [1948](https://github.com/psf/black/issues/1948) because it fails when tried to parse them.

Seems it was fixed in [21.11b0](https://github.com/psf/black/releases/tag/21.11b0), we are using 21.6b0, should we update Black?